pkg/util/healthcheckparser/util.go

```diff
@@ -0,0 +1,37 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package healthcheck provides healthcheck URL generators and parsers
+package healthcheckparser // import "k8s.io/kubernetes/pkg/util/healthcheckparser"
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+)
+
+func GenerateURL(namespace, name string) string {
+	return fmt.Sprintf("/healthchecks/svc/%s/%s", namespace, name)
```

Comment:
I think the concern here is do all loadbalancers support multiple `/` in the url? (I don't know but I suspect they do), and if I have a really long ns/svc, how do we deal with that? (i.e we can truncate that have no guarantee that it won't collide). 

A way around embedding the ns/svc-name in the url would be to seed the serving goroutine with the ns/svc-name when you spin it off (probably by storing it in proxy hc: https://github.com/kubernetes/kubernetes/pull/30569/files#diff-ca3cf2d117192ec03a15b4b1f2568e99R59).